pulser-pasqal/pulser_pasqal/job_parameters.py

```diff
@@ -15,15 +15,51 @@
 from __future__ import annotations
 
 import dataclasses
-from typing import Mapping, Union
+from typing import Dict, Mapping, Optional, Union
 
 from numpy.typing import ArrayLike
 
 from pulser.register import QubitId
 
+JobVariablesDict = Dict[str, Union[ArrayLike, Optional[Mapping[QubitId, int]]]]
+
+
+class JobVariables:
+    """Variables to build the sequence."""
+
+    def __init__(
+        self,
+        qubits: Optional[Mapping[QubitId, int]] = None,
+        **vars: Union[ArrayLike, float, int, str],
```

Comment:
We actually don't support `str` variables anymore, the `Sequence.build()` typehint is outdated. I think it's worth changing in both places